apps/site/src/_pages/docs/4_spec/04_service-module.mdx

```diff
@@ -0,0 +1,50 @@
+# Service Module
+
+This document (version 0.1) defines a **service module** according to the terms set out in the [core specification](https://blockprotocol.org/docs/spec/core).
+
+The service module enables blocks to interact with external services without needing to be provided with credentials: the embedding application calls the external service on the block's behalf.
+
+## Glossary
+
+**Provider:** a company, organization or person that provides a service or services (e.g. Mapbox)
+
+**Service:** a request handler which meets a specific functional goal, typically external to the embedding application and accessed via third-party API endpoint or integration (e.g. Retrieve Address).
+
+## Messages
+
+### Handling messages
+
+### <Hidden>Handling messages </Hidden>Summary
+
+The service module specifies named messages from named service providers, and any messages exchanged between application and block should be consistent with the shape and intent of those messages.
+
+How exactly the embedding application generates those messages is beyond the scope of the Block Protocol, so long as the schema and intent are met.
+
+#### <Hidden>Handling messages </Hidden>Specification
+
+Embedding applications MUST ensure that any responses they provide to blocks to messages sent under the service module are consistent with the schema and described intent of the message in this specification, and consistent with the response that the named service provider would return, except for any fields which are required by the named service provider but omitted from this specification (e.g. because they relate to authentication or other information that the block should not or does not have access to, and is added by the embedding application when making an onward request).
+
+They MAY implement messages by any of:
+
+- calling the named service provider directly, OR
+- generating the response internally, OR
+- calling another third-party that provides an appropriate response, whether directly or by themselves making a request to the named service provider.
+
+### Error definitions
+
+#### <Hidden>Error definitions </Hidden>Specification
+
+If a request fails, embedding applications MUST provide blocks with an error which complies with the specification, i.e. contains a listed code and a message.
+
+Embedding applications MAY withhold or translate error codes and messages sent by the service provider when reporting an error to the block.
+
+Blocks MUST NOT rely on error codes and messages providing information that the named service provider would if contacted directly, e.g. relating to billing.
+
+### Message definitions
+
+#### <Hidden>Message definitions </Hidden>Specification
+
+The messages available for exchange in the service module are defined in the [module's JSON definition](https://github.com/blockprotocol/blockprotocol/blob/main/libs/%40blockprotocol/service/src/service-module.json#L5) and are also
+listed below for ease of reference.
```

Comment:
These sections are inconsistent with the respective ones in the other module specifications.

(Both in terms of headings, and the description of the sections)

Reply:
How exactly? This looks identical to the equivalent section in the other specifications. Do you mean because there wasn't a 'Summary'? I've added one in https://github.com/blockprotocol/blockprotocol/pull/1049/commits/d592fd0bb94e4b53a8ed253f5478086b62248045, not sure it conveys much (I think other stuff about 'you can suggest adding things here' feels like it belongs elsewhere)

graph
```md
#### <Hidden>Message definitions </Hidden>Specification

The messages available for exchange in the graph module are defined in the [module's JSON definition](https://github.com/blockprotocol/blockprotocol/blob/main/libs/%40blockprotocol/graph/src/graph-module.json#L5) and are also
listed below for ease of reference.
```

hook
```
### <Hidden>Message definitions </Hidden>Specification

The messages available for exchange in the hook module are defined in the [module's JSON definition](https://github.com/blockprotocol/blockprotocol/blob/main/libs/%40blockprotocol/hook/src/hook-module.json#L5) and are also
listed below for ease of reference.
```

Reply:
I don't think it's particularly important, but you can see the inconsistent nestings here:
<img width="301" alt="image" src="https://user-images.githubusercontent.com/25749103/223397099-51064caa-461c-4def-a46c-8198be058feb.png">

Reply:
https://github.com/blockprotocol/blockprotocol/pull/1049/commits/42ec91d2b360f5c7a56e89d24ba5646bc968c7ae